### PR TITLE
zero-mAP fix remove `torch.empty()` forward pass in `.train()` mode

### LINKED ISSUE #9012: Using .onnx file for the test, all the metrics are 0(There is no problem with using .pt files)
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Using .onnx file for the test, all the metrics are 0 (There is no problem with using .pt files)
 Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100%|██████████| 253/253 [00:52<00:00,  4.85it/s]                                                                                                                                                                     
 all             253                 0               0          0            0                 0

### Additional

_No response_

### LINKED ISSUE #9038: Precision and Recall always be 0.
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

I run the template from the website“python train.py --data coco128.yaml --weights yolov5s.pt --img 640”, however the precision and recall has always been 0. I am sure I pull the newest version which means the train.py has been modified.
截屏2022-08-19 23 34 41

### Environment

exactly as same as the requirements.txt

### Minimal Reproducible Example

no

### Additional

 no

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

### LINKED ISSUE #9045: mAP always 0
### Search before asking

- [ ] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I Have been trying to custom train YOLOv5 model with one label and I have made suitable changes in YAML file .However, 
mAP value is still 0 no matter how many epochs I run the training for. I have been struggling with this issue for many days. 
Can someone suggest a solution?
Screenshot 2022-08-20 141130

### Additional

_No response_

### LINKED ISSUE #9059: Zero-mAP bug: P, R and mAP are all 0.0
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I train my custom one class dataset（about 40,000 pictures）use default cfg of yolov5s with the latest version（6.2）,P, R and mAP are all 0，(It's the same after about 100 epochs)however, it is normal to use the previous version（6.1）and it's normal when I train with coco128,I think this may be a bug, but I didn't debug it.

### Additional

_No response_

### PR DESCRIPTION
**Pull request by 0zppd**
Signed-off-by: 0zppd <111682241+0zppd@users.noreply.github.com>

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Upgraded logging to use zero-initialized tensor for improved stability during TensorBoard graph generation.

### 📊 Key Changes
- Changed input image initialization from `torch.empty()` to `torch.zeros()`.

### 🎯 Purpose & Impact
- Ensures more consistent initialization by using zero-filled tensors, which can help avoid any potential issues caused by uninitialized memory.
- The change could lead to more reliable model graph visualization in TensorBoard, benefiting users who analyze model architectures and debug them.

### CODE REVIEW

**Comment by AyushExel** [comment]
I think https://github.com/ultralytics/yolov5/pull/9067 includes these changes

**Comment by glenn-jocher** [comment]
@0zppd thanks for the PR! Are you sure this is the only place that torch.empty() is causing problems? See https://github.com/ultralytics/yolov5/pull/9067

**Comment by glenn-jocher** [comment]
@AyushExel if this is the only source of error then this PR is preferred as #9067 introduces some slight speed improvements, torch.empty() profiles faster than torch.zeros(), which was the original purpose behind #9067, but it seems the danger is when the model BN layers are in train mode and they are updating based on the inputs.

**Comment by glenn-jocher** [comment]
Confirmed on 10x colab training all good Screenshot 2022-08-21 at 15 40 00

**Comment by glenn-jocher** [comment]
@0zppd PR is merged. Thank you for your contributions to YOLOv5 🚀 and Vision AI ⭐

**Comment by robotwhispering** [comment]
Great work! You guys handled the business!

**Comment by pourmand1376** [comment]
I tested this PR on colab 30 times and I found no problem. I think this is finally resolved.

**Comment by glenn-jocher** [comment]
@pourmand1376 that's fantastic news! I'm glad to hear that the changes have been thoroughly tested and are proving effective. Your input and testing are invaluable to the YOLOv5 community. Thank you for your dedication to improving the project!
